frontend/src/components/cases/case-attachments-section.tsx

```diff
@@ -0,0 +1,735 @@
+  // Max attachments exceeded error fields
+  current_count?: number
+  max_count?: number
+  // Storage limit exceeded error fields
+  current_size_mb?: number
+  new_file_size_mb?: number
+  max_size_mb?: number
+}
+
+interface ApiErrorBody {
+  detail?: ApiErrorDetail | string
+}
+
+function getFileIcon(contentType: string) {
+  if (contentType.startsWith("image/")) return <ImageIcon className="h-4 w-4" />
+  if (contentType === "application/pdf") return <FileText className="h-4 w-4" />
+  if (contentType.includes("spreadsheet"))
+    return <FileSpreadsheet className="h-4 w-4" />
+  if (contentType.includes("presentation"))
+    return <Presentation className="h-4 w-4" />
+  if (contentType.startsWith("audio/")) return <Music className="h-4 w-4" />
+  if (contentType.startsWith("video/")) return <Video className="h-4 w-4" />
+  return <FileIcon className="h-4 w-4" />
+}
+
+function getFileColor(contentType: string) {
+  if (contentType.startsWith("image/")) return "text-green-600 bg-green-50"
+  if (contentType === "application/pdf") return "text-red-600 bg-red-50"
+  if (contentType.includes("spreadsheet"))
+    return "text-emerald-600 bg-emerald-50"
+  if (contentType.includes("presentation"))
+    return "text-orange-600 bg-orange-50"
+  if (contentType.startsWith("audio/")) return "text-purple-600 bg-purple-50"
+  if (contentType.startsWith("video/")) return "text-blue-600 bg-blue-50"
+  return "text-gray-600 bg-gray-50"
+}
+
+function formatFileSize(bytes: number): string {
+  if (bytes === 0) return "0 Bytes"
+  const k = 1024
+  const sizes = ["Bytes", "KB", "MB", "GB"]
+  const i = Math.floor(Math.log(bytes) / Math.log(k))
+  return Number.parseFloat((bytes / Math.pow(k, i)).toFixed(1)) + " " + sizes[i]
+}
+
+function truncateHash(hash: string): string {
+  return hash.substring(0, 8)
+}
+
+function getUploaderName(creatorId: string | null | undefined): string {
+  if (!creatorId) return "Unknown"
+  return "User"
+}
+
+// Add constant for max file size (20MB in bytes)
+const MAX_FILE_SIZE_BYTES = 20 * 1024 * 1024 // 20MB
+
+export function CaseAttachmentsSection({
+  caseId,
+  workspaceId,
+}: CaseAttachmentsSectionProps) {
+  const [isUploading, setIsUploading] = useState(false)
+  const [isDragOver, setIsDragOver] = useState(false)
+  const [previewAttachment, setPreviewAttachment] =
+    useState<CaseAttachmentRead | null>(null)
+  const [previewImageUrl, setPreviewImageUrl] = useState<string | null>(null)
+  const fileInputRef = useRef<HTMLInputElement>(null)
+  const queryClient = useQueryClient()
+
+  // Fetch attachments from API
+  const {
+    data: attachments = [],
+    isLoading: attachmentsLoading,
+    error: attachmentsError,
+  } = useQuery<CaseAttachmentRead[], ApiError>({
+    queryKey: ["case-attachments", caseId, workspaceId],
+    queryFn: async () => await casesListAttachments({ caseId, workspaceId }),
+  })
+
+  // Upload attachment mutation
+  const uploadMutation = useMutation({
```

Comment:
would be good to keep hooks separate. keeping data fetching separate from presentation